src/history/HistoryManagerImpl.cpp

```diff
@@ -806,6 +806,17 @@ HistoryManagerImpl::setPublicationEnabled(bool enabled)
 }
 #endif
 
+void
+HistoryManagerImpl::waitForCheckpointPublish()
+{
+    auto& clock = mApp.getClock();
+    while (!clock.getIOContext().stopped() && mPublishWork &&
+           !mPublishWork->isDone())
```

Comment:
I'm not certain this works 100% since the publish flow is async, so the guarantees about mPublishWork being set are a bit fuzzy. Also `isDone` includes failed attempts, which I don't think we want? I'd suggest using `HistoryManager::publishQueueLength() == 0` instead, as it guarantees that the checkpoint was successfully published.

Reply:
Done, I recall I had some issues with using the queue length, but it works now, so probably the issue was elsewhere and has been fixed.